runtime/src/in_mem_accounts_index.rs

```diff
@@ -63,6 +66,14 @@ pub enum InsertNewEntryResults {
     ExistedNewEntryNonZeroLamports,
 }
 
+#[derive(Default, Debug)]
+struct StartupInfo<T: IndexValue> {
+    /// entries to add next time we are flushing to disk
+    insert: Vec<(Slot, Pubkey, T)>,
+    /// pubkeys that were found to have duplicate index entries
+    duplicates: Vec<(Slot, Pubkey)>,
+}
```

Comment:
there are repeated calls to `insert_new_if_missing_into_primary_index` from many threads.
Meanwhile, the bg flusher threads are cycling through the bins, processing the inserts as fast as possible per bin.
Then, when `generate_index` is done, it sets startup to normal and waits for all flushes to complete.
Each time bg flusher threads process an insert, it could add a duplicate pubkey, while other threads are adding to `insert`.